### DISCUSSION TITLE
Possible Integration With AWS Cognito + ALB

### DISCUSSION THREAD

**Discussion by nitinkr0411** [open-webui/open-webui#3160] [Ideas]
Hi,

I'm working to make this compatible with Cognito User Pool attached with AWS Application Load Balancer with SAML federation in Azure AD.

Right now, oauth2-proxy does not handle SAML. Unless I'm dumb and don't know the right way to approach this.

But, in my mind, ALB sets a [x-amzn-oidc-data](https://docs.aws.amazon.com/elasticloadbalancing/latest/application/listener-authenticate-users.html#user-claims-encoding) header which is a json 
`{
   "sub": "1234567890",
   "name": "name",
   "email": "alias@example.com",
   ...
}`
that contains an email field.

Maybe we add an extra env [here](https://github.com/open-webui/open-webui/blob/162643a4b17edc24426980e4eddcdcfd7c3f7970/backend/config.py#L293) like
WEBUI_COGNITO_SAML_AUTH  

In that case, we just have to get the email from the jwt json and this would make it compatible with the setup.

Initially, I was thinking of maintaining my own fork, but I believe this could be a great addition for folks hosting the solution in enterprise scenarios.

Thoughts! Please feel free to correct me if I'm entirely wrong about this. :)

**Comment by pbpippen** [comment]
This would be great if you can get this working!

**Comment by nitinkr0411** [comment]
I have got this working in my fork. Happy to raise a PR if useful!
________________________________
From: pbpippen ***@***.***>
Sent: Saturday, June 15, 2024 10:25:46 PM
To: open-webui/open-webui ***@***.***>
Cc: Kumar, Nitin (RTX) ***@***.***>; Author ***@***.***>
Subject: [External] Re: [open-webui/open-webui] Possible Integration With AWS Cognito + ALB (Discussion #3160)


This would be great if you can get this working!

**Reply by pbpippen** [reply]
Yes, would be useful!

**Reply by jhcloobees** [reply]
We're interested in this too. @nitinkr0411 how did you go with the PR?

**Reply by user-error1** [reply]
Did this every getting integrated? Okta and other auth providers are fantastic, but for smaller orgs cognito could be great!

**Comment by Cabeda** [comment]
This would be great for my company too. Are there any updates on this? @nitinkr0411  did you ever get this working?
